apps/src/applab/manageAssets.jsx

```diff
@@ -0,0 +1,138 @@
+var React = require('react');
+var AssetsApi = require('./manageAssets/clientApi');
+var AssetRow = require('./manageAssets/assetRow.jsx');
+
+var errorMessages = {
+  415: 'This type of file is not supported.',
+  500: 'The server responded with an error.'
+};
+
+function getErrorMessage(status) {
+  return errorMessages[status] || 'An unknown error occurred.';
+}
+
+module.exports = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func,
+    typeFilter: React.PropTypes.string
+  },
+
+  getInitialState: function () {
+    return {
+      assets: null,
+      uploadStatus: ''
+    };
+  },
+
+  componentWillMount: function () {
+    // TODO: Use Dave's client api when it's finished.
+    AssetsApi.ajax('GET', '', function (xhr) {
+      var assets = JSON.parse(xhr.responseText);
+      if (this.props.typeFilter) {
+        assets = assets.filter(function (asset) {
+          return asset.category === this.props.typeFilter;
+        }.bind(this));
+      }
+      this.setState({assets: assets});
```

Comment:
It's possible that if the 'Upload Asset' button is clicked and its callback completes before this callback completes, that the newly uploaded asset would not appear in the list. One way to prevent this would be to disable the 'Upload Asset' button until this callback completes. I was not thinking about detecting duplicate names, but that is also a problem which would be solved by the same solution.